### ISSUE TITLE
All URLs are in use or disabled

### ISSUE THREAD

**Issue by robpur** [VorlonCD/bi-aidetection#120]
I've been running AI Tool and DS for some time with three cameras without much trouble, but after attempting to add a fourth camera I have been getting a message in the logs that all URLs are in use and processing stops until AI Tool is restarted.

Recently I did a two day test with four cameras saving images to aiinput, and four cameras defined in AI Tool, but the new camera was disabled in AI Tool. Everything worked fine until today when I checked the box in AI Tool to enable processing for the new camera. After that I was able to walk around in view of the three older cameras without an issue, but as soon as I walked in front of the new camera AI Tool reported that all URLs were in use, processing stopped, and all new images were just added to the queue. Restarting AI Tool will resolve the issue until the new camera sees motion, then processing stops again.

I'm running AI Tool 2.0 and the latest deepstack on a Jetson Nano.

Update: I tried the version you referenced in #118 but it has the same problem.

2020-12-27 13:46:24.246282|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134624199.jpg|  ====================== Adding new image to queue (Count=1): C:\AiInput\PumpHouse.20201227_134624199.jpg|491|1||8
2020-12-27 13:46:24.499127|Debug|AITOOLS.EXE|ImageQueueLoop|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134624199.jpg| Adding task for file 'PumpHouse.20201227_134624199.jpg' (Image QueueTime='0.0' mins, URL Queue wait='0ms', URLOrder=1 of 1, URLOriginalOrder=1) on URL 'http://127.0.0.1:5050/v1/vision/detection'|492|0||18
2020-12-27 13:46:24.499127|Debug|AITOOLS.EXE|ImageQueueLoop|127.0.0.1:5050|PumpHouse|None| Done adding 1 total threads, ErrCnt=0, ImageProcessQueue.Count=0|493|0||18
2020-12-27 13:46:24.499127|Debug|AITOOLS.EXE|DetectObjects|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134624199.jpg|  Starting analysis of C:\AiInput\PumpHouse.20201227_134624199.jpg...|494|1||15
2020-12-27 13:46:24.529110|Debug|AITOOLS.EXE|IsValidImage|127.0.0.1:5050|PumpHouse|None|   Image file is valid: PumpHouse.20201227_134624199.jpg|495|2||15
2020-12-27 13:46:24.531111|Debug|AITOOLS.EXE|DetectObjects|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134624199.jpg|  (1/6) Uploading a 937247 byte image to DeepQuestAI Server at http://127.0.0.1:5050/v1/vision/detection|496|1||15
2020-12-27 13:46:25.301644|Debug|AITOOLS.EXE|DetectObjects|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134624199.jpg|  (2/6) Posted in 770ms, Received a 126 byte response.|497|1||28
2020-12-27 13:46:25.301644|Debug|AITOOLS.EXE|DetectObjects|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134624199.jpg|  (3/6) Processing results...|498|1||28
2020-12-27 13:46:25.301644|Debug|AITOOLS.EXE|DetectObjects|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134624199.jpg|     Detected objects:|499|1||28
2020-12-27 13:46:25.302642|Debug|AITOOLS.EXE|Outsidemask|127.0.0.1:5050|PumpHouse|None|        ->Camera has no mask, the object is OUTSIDE of the masked area.|500|3||28
2020-12-27 13:46:25.302642|Info|AITOOLS.EXE|DetectObjects|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134624199.jpg|      Result='Relevant', Detail='Person (99.39%)', ObjType='Person', DynMaskResult='Unknown', DynMaskType='Unknown', ImgMaskResult='NoMaskImageFile', ImgMaskType='None'|501|1|Red|28
2020-12-27 13:46:25.302642|Debug|AITOOLS.EXE|DetectObjects|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134624199.jpg|  The summary:Person (99.39%)|502|1||28
2020-12-27 13:46:25.302642|Debug|AITOOLS.EXE|DetectObjects|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134624199.jpg|  (5/6) Performing alert actions:|503|1||28
2020-12-27 13:46:28.247123|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|PumpHouse|None|       |504|6||28
2020-12-27 13:46:28.247123|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134628228.jpg|       ====================== Adding new image to queue (Count=1): C:\AiInput\PumpHouse.20201227_134628228.jpg|505|6||28
2020-12-27 13:46:28.590910|Info|AITOOLS.EXE|WaitForNextURL|127.0.0.1:5050|PumpHouse|None|     ---- All URL's are in use or disabled, waiting...|506|5||17
2020-12-27 13:46:32.259711|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|PumpHouse|None|       |507|6||8
2020-12-27 13:46:32.259711|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134632240.jpg|       ====================== Adding new image to queue (Count=1): C:\AiInput\PumpHouse.20201227_134632240.jpg|508|6||8
2020-12-27 13:46:36.311268|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|PumpHouse|None|       |509|6||28
2020-12-27 13:46:36.311268|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|PumpHouse|PumpHouse.20201227_134636300.jpg|       ====================== Adding new image to queue (Count=2): C:\AiInput\PumpHouse.20201227_134636300.jpg|510|6||28
2020-12-27 13:46:47.258728|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|PumpHouse|None|       |511|6||8
2020-12-27 13:46:47.258728|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|Driveway|Driveway.20201227_134647335.jpg|       ====================== Adding new image to queue (Count=3): C:\AiInput\Driveway.20201227_134647335.jpg|512|6||8
2020-12-27 13:46:51.302485|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|Driveway|None|       |513|6||28
2020-12-27 13:46:51.302485|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|Driveway|Driveway.20201227_134651384.jpg|       ====================== Adding new image to queue (Count=4): C:\AiInput\Driveway.20201227_134651384.jpg|514|6||28
2020-12-27 13:46:52.702639|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|Driveway|None|       |515|6||8
2020-12-27 13:46:52.702639|Debug|AITOOLS.EXE|OnCreated|127.0.0.1:5050|GarageMotion|GarageMotion.20201227_134652768.jpg|       ====================== Adding new image to queue (Count=5): C:\AiInput\GarageMotion.20201227_134652768.jpg|516|6||8

**Comment by VorlonCD** [comment]
The version I just posted in https://github.com/VorlonCD/bi-aidetection/issues/104 may behave better in this respect?

If not, I need to see the section of the log that shows the final error for the URL that caused it to be disabled

**Comment by robpur** [comment]
The problem has been solved and it was due to my own mistake. I had not edited the trigger URL to contain my credentials. Previously the trigger URL was entered in the main camera setup page, but now it's another layer down in Actions Settings. This is the first time I've configured a camera in a newer version of AI Tool and since the setting was out of sight it was also out of mind. :-)  The problem was found after someone on the forum said that they had a similar issue. They had an incorrect trigger URL and they also received the message about all URLs  being in use or disabled.

I looked at the version posted in #104 and it's good to see so much work being done. It's been a while since the last official release and I'm happy to know that AI Tool is continuing to be improved. Thanks for all your effort and hard work!
